### PR TITLE
Automatically keep Intl enums in sync

### PR DESCRIPTION
**Pull request by jackhorton**
Went down a bit of a macro rabbit hole while investigating another bug in Intl; this approach is a (theoretically) safer version of what we were already doing. Also, I just _really_ wanted yet another bytecode-format-changing PR opened.

### CODE REVIEW

File: lib/Runtime/Base/JnDirectFields.h [hunk 1]
```diff
@@ -517,6 +517,31 @@ ENTRY(tagPublicLibraryCode)
...
+ENTRY(decimal)
+ENTRY(percent)
+ENTRY(NumberFormatStyle)
+ENTRY(NumberFormatCurrencyDisplay)
+ENTRY(CollatorSensitivity)
+ENTRY(CollatorCaseFirst)
+ENTRY(LocaleDataKind)
+
+// This symbol is not part of the regular Symbol API and is only used in rare circumstances in Intl.js for backwards compatibility
+// with the Intl v1 spec. It is visible to the user only using Object.getOwnPropertySymbols(Intl.NumberFormat.call(new Intl.NumberFormat())).
```
Comment (jackhorton) [inline lib/Runtime/Base/JnDirectFields.h]: these are remnants from my other PR, which this was originally based off of  but I wanted to keep it separate -- I will merge the other before this.

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp [hunk 2]
```diff
@@ -547,6 +566,22 @@ namespace Js
 #include "IntlExtensionObjectBuiltIns.h"
 #undef INTL_ENTRY
 
+        library->AddMember(intlNativeInterfaces, PropertyIds::FallbackSymbol, library->CreateSymbol(BuiltInPropertyRecords::_intlFallbackSymbol));
+
+        DynamicObject * enumObj = nullptr;
+
+// Projects the exact layout of our C++ enums into Intl.js so that we dont have to remember to keep them in sync
+#define ENUM_VALUE(enumName, unicodeKey, value) library->AddMember(enumObj, PropertyIds::##unicodeKey, JavascriptNumber::ToVar(value, scriptContext));
```
Comment (jackhorton) [inline lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: `unicodeKey` should really be `propId` or something similar -- I will change it after I rebase.

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp [hunk 3]
```diff
@@ -48,48 +48,67 @@ using namespace PlatformAgnostic::ICUHelpers;
 
 #endif // INTL_ICU
 
-// NOTE(jahorto): Keep these enums in sync with those by the same name in Intl.js
-// These enums are used by both WinGlob- and ICU-backed Intl
-enum class NumberFormatStyle
-{
-    Decimal, // Intl.NumberFormat(locale, { style: "decimal" }); // aka in our code as "number"
```
Comment (MSLaguana) [inline lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: Is it worth preserving these comments about the meaning/usage of the value?

Comment (jackhorton) [reply lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: _Meh._. I don't think they were particularly descriptive before.

Comment (jackhorton) [reply lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: I can add them back if others feel differently.

**Comment by MSLaguana** [comment]
Apparently this breaks a ton of TTD, and also some intl stuff:

```
Test threw exception: ReferenceError: 'NumberFormatStyle' is not defined
4>	at test (Collator.js:14:17)
4>	at body (Collator.js:18:13)
4>	at runTest (UnitTestFramework.js:261:17)
4>	at runTests (UnitTestFramework.js:185:21)
4>	at Global code (Collator.js:8:1)
```

**Comment by MSLaguana** [comment]
I think you missed some usages of these properties, e.g. https://github.com/Microsoft/ChakraCore/blob/b8c68078156e510b1e122163a64bf777d7baaafd/lib/Runtime/Library/InJavascript/Intl.js#L3000-L3005

**Comment by jackhorton** [comment]
Yeah, as always I forget that Intl-WinGlob exists. Ill fix those along with the rebase.

**Comment by jackhorton** [comment]
closing in favor of a slightly more expansive PR to minimize number of bytecode regens.

**Pull request closed without merging**